### ISSUE TITLE
Don't start the bridge if chainId cannot be obtained

### ISSUE THREAD

**Issue by akolotov** [poanetwork/bridge-nodejs#17]
Currently, the bridge gets ChainID for networks as part of the startup sequence.
https://github.com/poanetwork/bridge-nodejs/blob/a23ecb78572f3e1928d9129eff8d4118b2618b57/src/start.js#L13-L22

But if the ChainID cannot be obtained (e.g. due to accessibility of the node through RPC) the bridge continue operation even if it does not make sense since the chain id is used to sign bridge transactions. 

That's why the behavior must be changes as so to shutdown startup sequence if ChainID cannot be obtained.

Here is logs which are generated when the issue is reproduced:
```
Error! { FetchError: request to http://127.0.0.1:48545/ failed, reason: connect ECONNREFUSED 127.0.0.1:48545
    at ClientRequest.<anonymous> (/home/koal/git-repos/bridge-nodejs/node_modules/node-fetch/lib/index.js:1393:11)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at Socket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  message: 'request to http://127.0.0.1:48545/ failed, reason: connect ECONNREFUSED 127.0.0.1:48545',
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED' }
Home Chain ID: 64206
Foreign Chain ID: undefined
Found 0 Deposits on Home Network
writing db deposits 105
Found 0 CollectedSignatures on Home Network
writing db colsigs 105
Error! { FetchError: request to http://127.0.0.1:48545/ failed, reason: connect ECONNREFUSED 127.0.0.1:48545
    at ClientRequest.<anonymous> (/home/koal/git-repos/bridge-nodejs/node_modules/node-fetch/lib/index.js:1393:11)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at Socket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  message: 'request to http://127.0.0.1:48545/ failed, reason: connect ECONNREFUSED 127.0.0.1:48545',
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED' }
```

**Comment by rstormsf** [comment]
agree. the bridge should crash  with an error

**Comment by patitonar** [comment]
This issue was resolved as part of the new bridge architecture on #25 
If the Sender gets an error while trying to get `ChainId` it will crash.

https://github.com/poanetwork/bridge-nodejs/blob/819aa90993237e4c8a19ab5dfb6a406a492322ae/src/sender.js#L26-L37

**Comment by akolotov** [comment]
Thanks! @patitonar could you share example of the log message? My worry is whether a stack trace is displayed as is. We already have #23 and #16 to optimize the error messages.

**Comment by patitonar** [comment]
Stack trace are displayed, also error message is not very helpful. I think we can optimize the error message to be more explicit and remove the stack trace
```
Error: Invalid JSON RPC response: ""
    at Object.InvalidResponse (/home/altoros/repos/bridge/bridge-nodejs/node_modules/web3-core-helpers/src/errors.js:42:16)
    at XMLHttpRequest.request.onreadystatechange (/home/altoros/repos/bridge/bridge-nodejs/node_modules/web3-providers-http/src/index.js:73:32)
    at XMLHttpRequestEventTarget.dispatchEvent (/home/altoros/repos/bridge/bridge-nodejs/node_modules/xhr2/lib/xhr2.js:64:18)
    at XMLHttpRequest._setReadyState (/home/altoros/repos/bridge/bridge-nodejs/node_modules/xhr2/lib/xhr2.js:354:12)
    at XMLHttpRequest._onHttpRequestError (/home/altoros/repos/bridge/bridge-nodejs/node_modules/xhr2/lib/xhr2.js:544:12)
    at ClientRequest.<anonymous> (/home/altoros/repos/bridge/bridge-nodejs/node_modules/xhr2/lib/xhr2.js:414:24)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at Socket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by akolotov** [comment]
@patitonar let's create a separate issue for this error message.
by doing this we could close #17 since it was addressed by #25.

**Comment by pablofullana** [comment]
- [ ] Enhance error message in the fashion of "ChainID cannot be obtained for <home|foreign> side."
